Makefile.targets.mk

```diff
@@ -12,24 +12,26 @@ REV=$(shell git rev-parse --short HEAD)
 # Tag is the tag to use for build and push image targets.
 TAG ?= $(REV)
 
-.PHONY: build
-build:  ## Build the envoy-gateway binary
-	@CGO_ENABLED=0 go build -a -o ./bin/${GOOS}/${GOARCH}/ github.com/envoyproxy/gateway/cmd/envoy-gateway
+SUPPORT_ARCHS ?= amd64 arm64
+# Generate three targets: build-amd64, build-arm64
+# Run make build -n to see the result with dry run.
+BUILD_BINARY_ARCHS = $(addprefix build-,$(SUPPORT_ARCHS))
 
-build-linux-amd64:
-	@GOOS=linux GOARCH=amd64 $(MAKE) build
+.PHONY: build $(BUILD_BINARY_ARCHS)
+build: $(BUILD_BINARY_ARCHS) ## Build the envoy-gateway binary.
+$(BUILD_BINARY_ARCHS): build-%:
```

Comment:
this breaks `darwin` binaries

Reply:
Now `make build` replaces `make build-all` as `build-all` only support Linux, I will update for darwin,

Reply:
you might end up building something similar to what @Xunzhuo  already did in https://github.com/envoyproxy/gateway/pull/113, lets wait for @Xunzhuo to wrap up https://github.com/envoyproxy/gateway/pull/113 and then see if any more optimizations are needed.
thanks !

Reply:
Now:

```
root:[gateway]$ MAKE_IN_DOCKER=0 make build -n
make -f Makefile.targets.mk build
make[1]: Entering directory '/root/github/gateway'
CGO_ENABLED=0 GOOS="linux" GOARCH="amd64" go build -a -o ./bin/linux/amd64/ github.com/envoyproxy/gateway/cmd/envoy-gateway
CGO_ENABLED=0 GOOS="linux" GOARCH="arm64" go build -a -o ./bin/linux/amd64/ github.com/envoyproxy/gateway/cmd/envoy-gateway
CGO_ENABLED=0 GOOS="darwin" GOARCH="amd64" go build -a -o ./bin/linux/amd64/ github.com/envoyproxy/gateway/cmd/envoy-gateway
CGO_ENABLED=0 GOOS="darwin" GOARCH="arm64" go build -a -o ./bin/linux/amd64/ github.com/envoyproxy/gateway/cmd/envoy-gateway
make[1]: Leaving directory '/root/github/gateway'
```

Reply:
> you might end up building something similar to what @Xunzhuo already did in #113, lets wait for @Xunzhuo to wrap up #113 and then see if any more optimizations are needed. thanks !

Sure, I can wait.

Reply:
@arkodg seems my implementation looks easier to maintain, please take a look.